codejail/tests/test_safe_exec.py

```diff
@@ -38,6 +42,14 @@ def test_set_values(self):
         self.safe_exec("a = 17", globs)
         self.assertEqual(globs['a'], 17)
 
+    def test_complex_globals(self):
+        globs = {}
+        self.safe_exec(
+            "from builtins import bytes; test_dict = {1: bytes('a', 'utf8'), 2: 'b', 3: {1: bytes('b', 'utf8'), 2: (1, bytes('a', 'utf8'))}}",
+            globs
+        )
+        self.assertDictEqual(globs['test_dict'], {'1': 'a', '2': 'b', '3': {'1': 'b', '2': [1, 'a']}})
```

Comment:
@feanil  @nedbat The issue with the test was that when we convert dict to json it converts the set obj to list and also when convert it back it keeps them as list obj. It also when converting back from json to dict changes the type of dict keys to string. to fix the test i have updated the comparison value to check for dict with string keys and list insted of test value. Is it the right way or do we need to write the custom encoder and decoder to get the functionality we want.

Reply:
@Ayub-Khan I think that's fine for now.  If we find that there are issues up-stream with how we did this, we can come back and fix later.